DiscordBot/Modules/UserModule.cs

```diff
@@ -1,231 +1,525 @@
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.IO;
 using System.Linq;
 using System.Net;
-using System.Net.Http;
 using System.Text;
-using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using Discord;
 using Discord.Commands;
-using Discord.WebSocket;
 using DiscordBot.Extensions;
-using DiscordBot.Properties;
 using DiscordBot.Services;
 using DiscordBot.Settings.Deserialized;
 using HtmlAgilityPack;
-using Newtonsoft.Json;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace DiscordBot.Modules
 {
     public class UserModule : ModuleBase
     {
-        private readonly ILoggingService _loggingService;
+        private static List<string> _commandList;
+        
+        private static Settings.Deserialized.Settings _settings;
+        private readonly CurrencyService _currencyService;
         private readonly DatabaseService _databaseService;
-        private readonly UserService _userService;
         private readonly PublisherService _publisherService;
-        private readonly UpdateService _updateService;
-        private readonly CurrencyService _currencyService;
 
         private readonly Rules _rules;
-        private static Settings.Deserialized.Settings _settings;
-
-        public UserModule(ILoggingService loggingService, DatabaseService databaseService, UserService userService,
-            PublisherService publisherService, UpdateService updateService, CurrencyService currencyService,
-            Rules rules, Settings.Deserialized.Settings settings)
+        private readonly UpdateService _updateService;
+        private readonly UserService _userService;
+        
+        public UserModule(IServiceProvider serviceProvider, Settings.Deserialized.Settings settings, Rules rules)
         {
-            _loggingService = loggingService;
-            _databaseService = databaseService;
-            _userService = userService;
-            _publisherService = publisherService;
-            _updateService = updateService;
-            _currencyService = currencyService;
+            _databaseService = serviceProvider.GetService<DatabaseService>();
```

Comment:
I'm not 100% sold on using serviceProvider here, it works and as far as I can tell doesn't cause any problems but I couldn't find examples of it being used in modules so I'm not sure if a race condition could happen here.